pdl_syntax.md

```diff
@@ -0,0 +1,843 @@
+
+```
+namespace com.linkedin.pegasus.generator.examples
+
+/**
+ * Bar includes fields of Foo, Bar will have fields b1 from itself and f1 from Foo
+ */
+record Bar includes Foo {
+  b1: string
+}
+```
+```
+namespace com.linkedin.pegasus.generator.examples
+
+record Foo {
+  f1: string
+}
+```
+
+Multiple includes example:
+```
+namespace com.linkedin.pegasus.generator.examples
+
+/**
+ * Bar includes fields of Foo and Simple, Bar will have fields b1 from itself, f1 from Foo and s1 from Simple
+ */
+record Bar includes Foo, Simple {
+  b1: string
+}
+```
+```
+namespace com.linkedin.pegasus.generator.examples
+
+record Simple {
+  s1: string
+}
+```
+
+In pegasus, field inclusion does not imply inheritance, it is merely a convenience to reduce duplication when writing schemas.
+
+
+### Properties
+Properties can be used to present arbitrary data and added to records, fields and enums.
+
+For example: 
+```
+@prop = "value"
+record Fruits {
+  @validate.regex.regex = "^(yes|no)$"
+  field: string
+}
+```
+#### Property values can be any valid JSON type:
+
+For example: 
+```
+@prop = 1
+```
+```
+@prop = "string"
+```
+```
+@prop = [1, 2, 3]
+```
+```
+@prop = { "a": 1", "b": { "c": true }}
+
+```
+#### Property values can also be empty: 
+
+If you don't indicate an explicity property value, it will resuilt in an implicity value of `true`.
+
+For example: 
+```
+namespace com.linkedin.pegasus.generator.examples
+
+@prop
+record Foo {
+  f1: string
+}
+```
+#### Property keys can be expressed as JSON:
+
+For example: 
+```
+@a = {
+  "b": {
+    "c": {
+      "d": {
+        "e": {
+          "f": false
+        }
+      }
+    }
+  }
+}
+```
+
+#### Property keys can be expressed as paths:
+
+The JSON style property key is complicated to write and read, so we provide a shorthand - the dot separate format to express the property keys.
+
+The following example is equivalent to the previous JSON example:
+
+```
+@a.b.c.d.e.f = false
+```
+
+### Escaping
+There are some keywords which are reserved in Pegasus. If you have to use them to define any names, you need to put them in backticks : ` `.
+
+#### Keyword Escaping
+
+```
+namespace com.linkedin.pegasus.generator.test.idl.escaping
+
+record PdlKeywordEscaping {
+  `namespace`: string
+  `record`: string
+  `null`: string
+  `enum`: string
+  recordName: record `record` { }
+}
+```
+
+#### Namespace/Package escaping
+
+```
+namespace com.linkedin.pegasus.generator.test.idl.escaping.`record`
+package com.linkedin.pegasus.generator.test.idl.escaping.override.`typeref`
+
+/**
+ * Ensures that the namespace and package are properly escaped at the root as well as in scoped named-type declarations.
+ */
+record NamespacePackageEscaping {
+  x: {
+    namespace com.x.y.z.`enum`
+    package com.a.b.c.`fixed`
+
+    record Foo {}
+  }
+}
+```
+
+#### Property key escaping 
+
+If you want Pegasus to treat property key name with dots as one string key, please use backticks to escape such string.
```

Comment:
Also explain that escaping property keys can be used to escape reserved keywords such as "namespace" or "record"

Reply:
Updated.